javascript/README.md

```diff
@@ -283,6 +297,179 @@ const { a, ...noA } = copy; // noA => { b: 2, c: 3 }
 
 - ESLint: [prefer-object-spread]
-- JSPerf: [Shallow Copy Objects][JSPerf: Shallow Copy Objects]
+- JSPerf: [Shallow Copy Objects]
+
+[⇧ top](#javascript-guide)
+
+---
+
+## Arrays
+
+### 3.1 Array Creation
+
+Use the literal syntax for array creation.
+
+> Why? Use of the `Array` constructor to construct a new array is generally discouraged in favor of array literal notation because of the single-argument pitfall and because the `Array` global may be redefined.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const items = new Array();
+```
+
+🎉 Yep! 🎉
+
+```js
+const items = [];
+```
+
+#### Resources
+
+- ESLint: [no-array-constructor]
+
+
+### 3.2 Adding Items To Arrays
+
+Use [Array.prototype.push()][Array.prototype.push] instead of direct assignment to add items to an array.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const someStack = [];
+someStack[someStack.length] = 'abracadabra';
+```
+
+🎉 Yep! 🎉
+
+```js
+const someStack = [];
+someStack.push('abracadabra');
+```
+
+#### Resources
+
+- JSPerf: [Adding Array Items]
+
+### 3.3 Array Shallow-Copy
+
+Use [array spread syntax][Array Literal Spread Syntax] `...` to shallow-copy arrays.
+
+> Why? Better overall performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+// Too slow
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const len = animals.length;
+const animalsCopy = [];
+let i;
+
+for (i = 0; i < len; i ++) {
+  animalsCopy[i] = animals[i];
+}
+
+// Works but is not preferred
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = animals.slice();
+```
+
+🎉 Yep! 🎉
+
+```js
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = [...animals];
+```
+
+#### Resources:
+
+- JSPerf: [Shallow Copy Arrays]
+
+### 3.4 Arrays From Iterables
+
+To convert an iterable object (e.g. [NodeList]) to an array, use [array spread syntax][Array Literal Spread Syntax] `...` instead of [Array.from].
+
+> Why? Better performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = Array.from(paragraphs);
+```
+
+🎉 Yep! 🎉
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = [...paragraphs];
+```
+
+#### Resources
+
+- ESLint: [prefer-spread]
+- JSPerf: [Arrays From Iterables]
+
+### 3.5 Arrays from Array-Like Objects
+
+Use [`Array.from`][Array.from] for converting an array-like object to an array.
+
+> Why? Not only is it easier to read/type but it also performs better.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const arrLike = { 0: 'foo', 1: 'bar', 2: 'baz', length: 3 };
+const arr = Array.prototype.slice.call(arrLike);
+```
+
+🎉 Yep! 🎉
+
+```js
+const arrLike = { 0: 'foo', 1: 'bar', 2: 'baz', length: 3 };
+const arr = Array.from(arrLike);
+```
+
+#### Resources
+
+- JSPerf: [Arrays from Array-Like Objects]
+
+### 3.6 Mapping Over Iterables
+
+Use [array spread syntax][Array Literal Spread Syntax] `...` instead of [`Array.from`][Array.from] for mapping over iterables.
```

Comment:
I ran a JSPerf and it seems going with the opposite of what the Airbnb Guide suggests performs better overall. I'd like your thoughts, if you have any, on this. I wanted to make sure my JSPerf tests were accurate and that I wasn't testing incorrectly.

Reply:
I tried it in Chrome 69 and Chrome 70 on Linux, which resulted in spread being more than twice as fast, and I also tried FF on Linux, where spread is almost twice as fast.

Reply:
I also created this test, which compared `Array.from()` and `[...].map()` as well as `Array.from().map`. https://jsperf.com/array-from-vs-spread-vs-array-from-map/1

Reply:
Nice! Thanks for the test! I think I'll add that to the docs! 👍

Reply:
Interesting. On this note, is there any thought of adding comments about native support (thinking browsers), so that we don't end up relying on Babel or polyfills to use what should be more performant, future-friendly methods that are actually being transpiled to something less performant, at least for some clients. I realize this is a moving target and also very much a case-by-case thing, but even links to [caniuse](https://caniuse.com) might be helpful.